File: services/device/src/main/java/com/example/iter/device/util/AdminEquipmentMapper.java
package com.example.iter.device.util;

import com.example.iter.auth.api.UserSummary;
import com.example.iter.device.domain.entity.Equipment;
import com.example.iter.device.domain.entity.EquipmentImage;
import com.example.iter.device.dto.response.AdminEquipmentDetailResponse;
import com.example.iter.device.dto.response.AdminEquipmentSummaryResponse;
import com.example.iter.device.dto.response.ImageResponse;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class AdminEquipmentMapper {

    // 장비, 등록자, 썸네일을 관리자 장비 목록 응답으로 변환합니다.
    public AdminEquipmentSummaryResponse toSummary(
            Equipment equipment,
            UserSummary owner,
            String thumbnailUrl
    ) {
        return new AdminEquipmentSummaryResponse(
                equipment.getId(),
                equipment.getName(),
                equipment.getCategory(),
                equipment.getDailyPrice(),
                equipment.getStatus(),
                owner,
                thumbnailUrl,
                equipment.getCreatedAt()
        );
    }

    // 장비, 등록자, 전체 이미지 목록을 관리자 장비 상세 응답으로 변환합니다.
    public AdminEquipmentDetailResponse toDetail(
            Equipment equipment,
            UserSummary owner,
            List<EquipmentImage> images
    ) {
        List<ImageResponse> imageResponses = images.stream().map(this::toImageResponse).toList();

        return new AdminEquipmentDetailResponse(
                equipment.getId(),
                owner,
                equipment.getCategory(),
                equipment.getName(),
                equipment.getDescription(),
                equipment.getDailyPrice(),
                equipment.getAvailableFrom(),
                equipment.getAvailableTo(),
                equipment.getStatus(),
                equipment.getProductCondition(),
                equipment.getConditionDetail(),
                imageResponses,
                equipment.getCreatedAt(),
                equipment.getUpdatedAt()
        );
    }

    // 장비 이미지 엔티티를 이미지 응답 DTO로 변환합니다.
    private ImageResponse toImageResponse(EquipmentImage image) {
        return new ImageResponse(
                image.getId(),
                image.getImageUrl(),
                image.getSortOrder(),
                image.getThumbnail()
        );
    }
}

File: services/device/src/main/kotlin/com/example/iter/device/util/AdminEquipmentMapper.kt
package com.example.iter.device.util

import com.example.iter.auth.api.UserSummary
import com.example.iter.device.domain.entity.Equipment
import com.example.iter.device.domain.entity.EquipmentImage
import com.example.iter.device.dto.response.AdminEquipmentDetailResponse
import com.example.iter.device.dto.response.AdminEquipmentSummaryResponse
import com.example.iter.device.dto.response.ImageResponse
import org.springframework.stereotype.Component

@Component
class AdminEquipmentMapper {

    // 장비, 등록자, 썸네일을 관리자 장비 목록 응답으로 변환합니다.
    // id 단언은 전부 같은 근거다 — 저장된 엔티티만 여기로 들어온다.
    fun toSummary(
        equipment: Equipment,
        owner: UserSummary,
        thumbnailUrl: String?,
    ): AdminEquipmentSummaryResponse = AdminEquipmentSummaryResponse(
        equipment.id!!,
        equipment.name,
        equipment.category,
        equipment.dailyPrice,
        equipment.status,
        owner,
        thumbnailUrl,
        equipment.createdAt,
    )

    // 장비, 등록자, 전체 이미지 목록을 관리자 장비 상세 응답으로 변환합니다.
    fun toDetail(
        equipment: Equipment,
        owner: UserSummary,
        images: List<EquipmentImage>,
    ): AdminEquipmentDetailResponse = AdminEquipmentDetailResponse(
        equipment.id!!,
        owner,
        equipment.category,
        equipment.name,
        equipment.description,
        equipment.dailyPrice,
        equipment.availableFrom,
        equipment.availableTo,
        equipment.status,
        equipment.productCondition,
        equipment.conditionDetail,
        images.map(::toImageResponse),
        equipment.createdAt,
        equipment.updatedAt,
    )

    // 장비 이미지 엔티티를 이미지 응답 DTO로 변환합니다.
    private fun toImageResponse(image: EquipmentImage): ImageResponse = ImageResponse(
        image.id!!,
        image.imageUrl,
        image.captureView,
        image.sortOrder,
        image.thumbnail
    )
}
